### PR TITLE
Chore/redux cleanup

### PR DESCRIPTION
**Pull request by ghost**
**Yes, I know, 40 files, but not actually much added, just restructuring**

**Why is this change needed?**
Cleaning up the redux will improve maintainability and extensibility. Helps to make the redux reducers more modular. Generally just more organised. Testing added improves reliability and provides a template for future redux tests to be copied off of.

**What did you change?**
Basically all of redux, sorry. Mostly just restructuring. Did edit the `currentProjectReducer` to fix a bug when handling payloads. Also, add tests, both for testing a reducer and also for testing actions. Finally, one or two changes to fix errors/warnings for ESLint, `yarn lint` is now much more clean.

**Note**
Had to move the `Action` type interface out of `sharedHelpers.ts` to avoid a circular dependency with the `ops` files.

Wondering if long term we need both reducer tests and actions tests (as action tests dispatch to reducer and then check reducer - hence, they encompass reducer tests). Maybe we leave them as a template???

**Manika** and **Riley**, I believe you are both using redux in your current/upcoming PRs. If this PR is merged in first then will need to allow for it.

### CODE REVIEW

File: src/renderer/store/currentProject/__tests__/actions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { createStore } from 'redux';
+import { mockProject } from '../../../../../mocks/mocks';
+import currentPageReducer from '../reducer';
```
Comment (chloebrett) [inline src/renderer/store/currentProject/__tests__/actions.test.tsx]: `currentProjectReducer`

File: src/renderer/store/currentProject/__tests__/reducer.test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import currentPageReducer from '../reducer';
```
Comment (chloebrett) [inline src/renderer/store/currentProject/__tests__/reducer.test.tsx]: Again

File: src/renderer/store/exportIo/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/**
+ * All input/output user actions states
+ * Import / Export progress states
+ */
+export interface ExportIO {
```
Comment (chloebrett) [inline src/renderer/store/exportIo/helpers.ts]: For consistency, treat acronyms as regular words in camelcase e.g. `exportIo`

**Review by chloebrett — CHANGES_REQUESTED** [review]
Looking like a massive improvement, just a few typos

File: src/renderer/store/exportIo/helpers.ts (hunk 3 above)
Comment (ghost) [reply src/renderer/store/exportIo/helpers.ts]: Yeah good idea. Thanks. Fixed

File: src/renderer/store/currentProject/__tests__/actions.test.tsx (hunk 1 above)
Comment (ghost) [reply src/renderer/store/currentProject/__tests__/actions.test.tsx]: Fixed

File: src/renderer/store/currentProject/__tests__/reducer.test.tsx (hunk 2 above)
Comment (ghost) [reply src/renderer/store/currentProject/__tests__/reducer.test.tsx]: Fixed

**Comment by ghost** [comment]
@rileykeane and @patrickbrett  merged in develop with changes from your PRs. Ran lint and now no errors. Please do a double check to make sure I haven't ruined anything special with incorrect import paths, etc

**Review by chloebrett — APPROVED** [review]
Looks good to me, both unit tests and manual testing passed on my end

**Pull request merged**